yjit/src/backend/arm64/mod.rs

```diff
@@ -175,7 +175,7 @@ impl Assembler
                         }
                     }
                 },
-                Op::And => {
+                Op::And | Op::Or => {
```

Comment:
By having `Op::Or` here, it seemed like `asm.or` may generate `and` instead of `orr`. I needed to touch this part in https://github.com/Shopify/ruby/pull/364 to pass the tests in the pull request.